src/In.ProjectEKA.HipService/Common/GatewayClient.cs

```diff
@@ -0,0 +1,70 @@
+namespace In.ProjectEKA.HipService.Common
+{
+    using System;
+    using System.Net.Http;
+    using System.Net.Mime;
+    using System.Text;
+    using Common;
+    using In.ProjectEKA.HipLibrary.Patient.Model;
+    using Model;
+    using Logger;
+    using Microsoft.Net.Http.Headers;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Serialization;
+    using Task = System.Threading.Tasks.Task;
+
+    public class GatewayClient
+    {
+        private readonly HttpClient httpClient;
+        private readonly CentralRegistryClient centralRegistryClient;
+
+        public GatewayClient(HttpClient httpClient, CentralRegistryClient centralRegistryClient)
+        {
+            this.httpClient = httpClient;
+            this.centralRegistryClient = centralRegistryClient;
+        }
+
+        public virtual async Task SendDataToGateway(String url, GatewayDiscoveryRepresentation discoveryResponse)
+        {
+            await PostTo(url, discoveryResponse);
+        }
+
+        private async Task PostTo(string gatewayUrl, GatewayDiscoveryRepresentation representation)
+        {
+            try
+            {
+                var token = await centralRegistryClient.Authenticate();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5550197)